wordpress-mobile/WordPress-Android
# Dismiss Suggestions on Space Keystroke

Partial fix for the issues outlined in https://github.com/WordPress/gutenberg/issues/24425

The iOS side to this issue can be found at https://github.com/wordpress-mobile/WordPress-iOS/pull/16601

## Description

If a user enters an `@` symbol on mobile, the suggestions UI automatically opens. There are times when a user doesn't want this UI but, instead, only wants to add a `@` symbol. This PR seeks to make it easier for users to exit the suggestions UI, by automatically closing it if they press a space key immediately after adding a `@`.

## Testing Steps

The testing steps match [the iOS side](https://github.com/wordpress-mobile/WordPress-iOS/pull/16601) of this issue:

* Verify that in the block editor, typing a space after a `@` character dismisses the mentions suggestions list
* Verify that in the block editor, typing a space after a `+` character dismisses the cross-posting suggestions list
* Verify that mentions still works, ensuring that the intended recipient gets notified
* Verify that cross-posting still works, ensuring that the intended recipient gets notified

In addition, this PR includes some [changes to the way Tracks events are called from `finishWithValue()`](https://github.com/wordpress-mobile/WordPress-Android/pull/14793/commits/9459b6cd841d4995c4b862da97261a320436b882). We should test that the event fires as expected, by following these steps: 

* Navigate to the **Live View** section of the Tracks tool. Enter the username of the account you're testing with and use search for the `wpandroid_suggestion_session_finished` event name. Note, it can take anywhere from five minutes to two hours for an event to be recorded.
* Verify that the `did_select_suggestion` is `true` in cases where you select the suggestion in your testing and `false` in cases where you exist the UI via the space keystroke.

P2 detailing the Tracks event, for reference: p9ugOq-1tR-p2

## Regression Notes
1. Potential unintended areas of impact

* Any new breaking or unusual behaviour with the existing suggestions UI.
* Incorrect values returning for the `did_select_suggestion` property in the Tracks event i.e. `true` when the user exits the UI without making a selection or `false` when they make a selection.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I followed the testing steps listed above.

3. What automated tests I added (or what prevented me from doing so)

As this is a small change and isn't a core flow, a UI test doesn't feel warranted.

PR submission checklist:
